torch/csrc/jit/codegen/cuda/compute_at.cpp

```diff
@@ -601,6 +601,39 @@ void ComputeAt::traverseForward() {
   }
 }
 
+void ComputeAt::resetMaxProducerPos(TensorView* consumer_tv) {
+  if (consumer_tv->definition() == nullptr) {
+    consumer_tv->setMaxProducer(0, true);
+  }
+
+  auto producer_tvs = ir_utils::producerTvsOf(consumer_tv);
+
+  unsigned int new_consummer_pa_pos = 0;
+
+  // Re-compute the max producer position as one or more
+  //  of the producers of this consumer have updated their
+  //  compute at position.
+  for (auto inp : ir_utils::producerTvsOf(consumer_tv)) {
+    if (!inp->isFusionInput()) {
+      // Locate consumer's position that aligns with
+      //  the producer's new compute at axis.
+      unsigned int inp_ca_pos_to_consumer =
+          getConsumerPosAlignedToProducerCA(consumer_tv, inp);
+
+      // Populate the max consumer position required by
+      //  producer compute at.
+      new_consummer_pa_pos =
+          std::max(new_consummer_pa_pos, inp_ca_pos_to_consumer);
+    }
+  }
+
+  // After going through all the producers, decrease the produce
+  //  position of current consumer if needed.
+  if (new_consummer_pa_pos <= consumer_tv->getMaxProducerPosition()) {
```

Comment:
Why doesn't this update the max position when the new position is larger than the previous one? I'm not sure if it actually could happen, but the new produce-at position seems to be the right value in any case.